From: Boyuan Yang <073plan@gmail.com>
Date: Fri, 20 Jul 2018 13:05:17 +0800
Subject: Use Qt5 lrelease / pylupdate5 instead of qt4 tools

---
 eyes17/lang/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/eyes17/lang/Makefile b/eyes17/lang/Makefile
index 8825a15..004b1e3 100644
--- a/eyes17/lang/Makefile
+++ b/eyes17/lang/Makefile
@@ -2,12 +2,12 @@ DESTDIR =
 TS_FILES = $(shell ls *.ts)
 QM_FILES = $(patsubst %.ts, %.qm, $(TS_FILES))
 SOURCES = $(shell ls ../*py ../code/*.py)
-PYLUPDATE ?= pylupdate4
+PYLUPDATE ?= pylupdate5
 
 all: $(QM_FILES)
 
 $(QM_FILES): $(TS_FILES)
-	lrelease-qt4 eyes17.pro
+	lrelease eyes17.pro
 
 $(TS_FILES): eyes17.pro
 	$(PYLUPDATE) eyes17.pro
