removed calls for python2
Index: expeyes/Makefile
===================================================================
--- expeyes.orig/Makefile
+++ expeyes/Makefile
@@ -4,7 +4,6 @@ SUBDIRS = bin po firmware clib/expeyes-c
 SUBDIRS_INDEP = expeyes-web eyes17/lang
 
 all:
-	python setup.py build
 	python3 setup.py build
 	for d in $(SUBDIRS); do \
 	  if [ -f $$d/configure.ac ]; then \
@@ -25,14 +24,11 @@ all_indep:
 	done
 
 install:
-	# for python-expeyes
+	# for python3-expeyes
 	if grep -Eq "Debian|Ubuntu" /etc/issue; then \
-	  python setup.py install --install-layout=deb \
-	         --root=$(DESTDIR)/ --prefix=/usr; \
 	  python3 setup.py install --install-layout=deb \
 	         --root=$(DESTDIR)/ --prefix=/usr; \
 	else \
-	  python setup.py install --root=$(DESTDIR)/ --prefix=/usr; \
 	  python3 setup.py install --root=$(DESTDIR)/ --prefix=/usr; \
 	fi
 	install -d $(DESTDIR)/lib/udev/rules.d/
Index: expeyes/expeyes-web/cgi-bin/expeyes_setoutput.cgi
===================================================================
--- expeyes.orig/expeyes-web/cgi-bin/expeyes_setoutput.cgi
+++ expeyes/expeyes-web/cgi-bin/expeyes_setoutput.cgi
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 """
 CGI script for setting status for output pins of expEYES Junior connected to server
 Copyright 2016,  Authors : , Jishnu R(jishnu47@gmail.com), Rakesh K M(rakeshkm2203@gmail.com), Manoj.S.Nair(manojsnair007@gmail.com) [Amrita School of Engineering, Amritapuri Campus, Kollam 690525, Kerala]
Index: expeyes/expeyes-web/cgi-bin/expeyes_status.cgi
===================================================================
--- expeyes.orig/expeyes-web/cgi-bin/expeyes_status.cgi
+++ expeyes/expeyes-web/cgi-bin/expeyes_status.cgi
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 """
 CGI script for reading status of input pins of expEYES Junior connected to server
 Copyright 2016,  Authors : Manoj.S.Nair(manojsnair007@gmail.com), Jishnu R(jishnu47@gmail.com), Rakesh K M(rakeshkm2203@gmail.com) [Amrita School of Engineering, Amritapuri Campus, Kollam 690525, Kerala]
