renewed rules after Ajith Kumar's email
Index: expeyes-4.3.6+dfsg/99-phoenix.rules
===================================================================
--- expeyes-4.3.6+dfsg.orig/99-phoenix.rules
+++ expeyes-4.3.6+dfsg/99-phoenix.rules
@@ -1,4 +1,9 @@
-# udev rules for expEYES interface: AVR, FT232 and MCP2200
-ATTR{idVendor}=="03eb", ATTR{idProduct}=="21ff", MODE="666"
-ATTR{idVendor}=="0403", ATTR{idProduct}=="6001", MODE="666"
-ATTR{idVendor}=="04d8", ATTR{idProduct}=="00df", MODE="666"
+# udev rules for expEYES interface: AVR, FT232, MCP2200 and CH340
+SUBSYSTEM=="usb",ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="21ff", MODE="666"
+SUBSYSTEM=="tty",ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6001", MODE="666"
+SUBSYSTEM=="tty",ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="00df", MODE="666"
+SUBSYSTEM=="tty",ATTRS{idVendor}=="1a86", ATTRS{idProduct}=="7523", MODE="666"
+ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="21ff", ENV{ID_MM_DEVICE_IGNORE}="1"
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6001", ENV{ID_MM_DEVICE_IGNORE}="1"
+ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="00df", ENV{ID_MM_DEVICE_IGNORE}="1"
+ATTRS{idVendor}=="1a86", ATTRS{idProduct}=="7523", ENV{ID_MM_DEVICE_IGNORE}="1"
